.github/workflows/ci.yml

```diff
@@ -0,0 +1,54 @@
+name: CI
+
+on:
+  push:
+    # branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  # Run daily at 1:23 UTC
+  schedule:
+  - cron:  '23 1 * * *'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: "test upload via action"
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install python-build and twine
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install build twine
+          python -m pip list
+
+      - name: Build a wheel and a sdist
+        run: |
+          PYTHONWARNINGS=error,default::DeprecationWarning python -m build tests/debug_project
+          mv tests/debug_project/dist .
+
+      - name: Verify the distribution
+        run: twine check --strict dist/*
+
+      - name: List contents of sdist
+        run: python -m tarfile --list dist/debug-*.tar.gz
+
+      - name: List contents of wheel
+        run: python -m zipfile --list dist/debug-*.whl
+
+      - name: Test upload
+        uses: ./
+        with:
+          artifacts_path: dist
+          anaconda_nightly_upload_token: ${{ secrets.UPLOAD_TOKEN }}
```

Comment:
Currently broken by https://github.com/Anaconda-Platform/anaconda-client/issues/676

Reply:
Resolved (ish) in https://github.com/Anaconda-Platform/anaconda-client/issues/676#issuecomment-1644930931.

Reply:
Unfortunately, `anaconda-client` doesn't know how to deal with valid metadata from non-setuptools PEP 518 backends. :(

https://github.com/Anaconda-Platform/anaconda-client/issues/676#issuecomment-1645011255

So this forces us to use `setuptools` for the time being, which is annoying.